addons/third-party-button/userscript.js

```diff
@@ -0,0 +1,30 @@
+﻿export default async function ({ addon, global, console }) {
+  addon.settings.addEventListener("change", () => console.log("changed!"));
+  const viewon = document.createElement("span");
+
+  viewon.innerHTML = "View on ";
+  viewon.title = "Third-party sites";
+  viewon.style = "font-size:15px;float:right;";
+  console.log(window.location.href);
+  var urllist = window.location.href.split("/");
+  var projectid;
+  for (var i = urllist.length; i >= 0 && !projectid; i--) {

```

Comment:
Interesting way to go through an URL lol
I would have just used `new URL(location.href).pathname.split("/")[2]`

Reply:
well you can use /projects/id instead of /projects/id/ and that would screw everything up so I just look for the last truthy part.